File: lib/model/land.dart
class Land {
  String landName;
  String location;
  String provinsi;
  String kota;
  String kecamatan;
  String kelurahan;
  String surfaceArea;
  String deskripsi;
  String rentalPrice;
  double rating;
  List<String> landPhotos;
  LandOwner landOwner;

  Land({
    required this.landName,
    required this.location,
    required this.provinsi,
    required this.kota,
    required this.kecamatan,
    required this.kelurahan,
    required this.surfaceArea,
    required this.deskripsi,
    required this.rentalPrice,
    required this.rating,
    required this.landPhotos,
    required this.landOwner,
  });

  factory Land.fromJson(Map<String, dynamic> json) {
    return Land(
      landName: json['land_name'],
      location: json['location'],
      provinsi: json['provinsi'],
      kota: json['kota'],
      kecamatan: json['kecamatan'],
      kelurahan: json['kelurahan'],
      surfaceArea: json['surface_area'],
      deskripsi: json['deskripsi'],
      rentalPrice: json['rental_price'],
      rating: json['rating'],
      landPhotos: List<String>.from(json['land_photos']),
      landOwner: LandOwner.fromJson(json['land_owner']),
    );
  }
}

class LandOwner {
  String name;
  String username;
  String email;
  String phoneNumber;

  LandOwner({
    required this.name,
    required this.username,
    required this.email,
    required this.phoneNumber,
  });

  factory LandOwner.fromJson(Map<String, dynamic> json) {
    return LandOwner(
      name: json['name'],
      username: json['username'],
      email: json['email'],
      phoneNumber: json['phone_number'],
    );
  }
}

File: src/service/land.js
import axios from 'axios';

const BaseURL = "http://127.0.0.1:8000/api/v1";

const LandService = {
  CheckAsLandowner: async (token) => {
    try {
      let config = {
        method: 'get',
        maxBodyLength: Infinity,
        url: `${BaseURL}/land/check/landowner`,
        headers: {
          Accept: 'application/json',
          Authorization: `Bearer ${token}`,
        },
      };

      const response = await axios.request(config);
      return response.data;
    } catch (error) {
      console.error('API error:', error);
      throw error; 
    }
  },
  
  RegisterAsLandowner: async(cred) => {
    try {
        const FormData = require('form-data');
        let data = new FormData();
        data.append('phone_number', cred.phone_number);
        data.append('owner_adress', cred.owner_address);

        let config = {
        method: 'post',
        maxBodyLength: Infinity,
        url: `${BaseURL}/land/landowner/register`,
        headers: { 
            Authorization: `Bearer ${cred.token}`, 
            Accept:"application/json"
        },
        data : data
        };

        const response = await axios.request(config);
        return response.data;
      } catch (error) {
        console.error('API error:', error);
        throw error; 
      }
  },
  CreateNewLand: async(cred) => {
    try {
        const FormData = require('form-data');
        let data = new FormData();
        data.append('land_name', cred.land_name);
        data.append('location', cred.location);
        data.append('surface_area', cred.surface_area);
        data.append('deskripsi', cred.deskripsi);
        data.append('rental_price',cred.rental_price);
        data.append('provinsi', cred.provinsi);
        data.append('kota', cred.kota);
        data.append('kecamatan', cred.kecamatan);
        data.append('kelurahan', cred.kelurahan);
        data.append('batas_tagihan', cred.batas_tagihan);
    
        let config = {
        method: 'post',
        maxBodyLength: Infinity,
        url: `${BaseURL}/land/create`,
        headers: { 
            Authorization: `Bearer ${cred.token}`, 
            Accept:"application/json"
        },
        data : data
        };
    
        const response = await axios.request(config);
        return response
    } catch (error) {
        throw error;
    }

  }
};

export default LandService;
